cpp/locality/NeighborQuery.h

```diff
@@ -265,7 +265,16 @@ class NeighborQueryPerPointIterator : public NeighborPerPointIterator
         : NeighborPerPointIterator(query_point_idx), m_neighbor_query(neighbor_query),
           m_query_point(query_point), m_finished(false), m_r_max(r_max), m_r_min(r_min),
           m_exclude_ii(exclude_ii)
-    {}
+    {
+        if (r_max <= 0)
+        {
+            throw std::invalid_argument("NeighborQuery requires r_max to be positive.");
+        }
+        if (r_max <= r_min)
+        {
+            throw std::invalid_argument("NeighborQuery requires that r_max must be greater than r_min.");
+        }
+    }
```

Comment:
Can we add Python tests for this? It probably needs to be tested using concrete subclasses like AABBQuery and LinkCell. We need to be sure we get `except +` on all the methods that can construct a NeighborQuery -- that might take some careful thought.

Reply:
Sure thing, I'll take care of this.

Reply:
@andkerr I changed the new tests a bit in ad4e6f4e9148796d7690fd749d85e812c2f2513d. It looks like you defined the tests in both AABBQuery and LinkCell classes, but the logic can be defined once in their shared parent class.